qiskit/providers/ibmq/api/clients/experiment.py

```diff
@@ -0,0 +1,257 @@
+
+        Returns:
+            Retrieved experiment plot.
+        """
+        return self.base_api.experiment_plot(experiment_id, plot_name).retrieve()
+
+    def experiment_plot_delete(self, experiment_id: str, plot_file_name: str) -> None:
+        """Delete an experiment plot.
+
+        Args:
+            experiment_id: Experiment UUID.
+            plot_file_name: Plot file name.
+        """
+        self.base_api.experiment_plot(experiment_id, plot_file_name).delete()
+
+    def experiment_devices(self) -> List:
+        """Return list of experiment devices.
+
+        Returns:
+            A list of experiment devices.
+        """
+        return self.base_api.experiment_devices()
+
+    def analysis_results(
+            self,
+            backend_name: Optional[str] = None,
+            device_components: Optional[List[str]] = None,
+            experiment_uuid: Optional[str] = None,
+            result_type: Optional[str] = None,
+            quality: Optional[List[str]] = None
+    ) -> List[Dict]:
+        """Return a list of analysis results.
+
+        Args:
+            backend_name: Name of the backend.
+            device_components: A list of device components used for filtering.
+            experiment_uuid: Experiment UUID used for filtering.
+            result_type: Analysis result type used for filtering.
+            quality: Quality value used for filtering.
+
+        Returns:
+            A list of analysis results.
+        """
+        return self.base_api.analysis_results(
+            backend_name, device_components, experiment_uuid, result_type, quality)
+
+    def analysis_result_upload(self, result: Dict) -> Dict:
+        """Upload an analysis result.
+
+        Args:
+            result: The analysis result to upload.
+
+        Returns:
+            Analysis result data.
+        """
+        return self.base_api.analysis_result_upload(result)
+
+    def analysis_result_update(self, result_id: str, new_data: Dict) -> Dict:
+        """Update an analysis result.
+
+        Args:
+            result_id: Analysis result ID.
+            new_data: New analysis result data.
+
+        Returns:
+            Analysis result data.
+        """
+        return self.base_api.analysis_result(result_id).update(new_data)
+
+    def analysis_result_delete(self, result_id: str) -> Dict:
+        """Delete an analysis result.
+
+        Args:
+            result_id: Analysis result ID.
+
+        Returns:
+            Analysis result data.
+        """
+        return self.base_api.analysis_result(result_id).delete()
+
+    def analysis_result_get(self, result_id: str) -> Dict:
+        """Retrieve an analysis result.
+
+        Args:
+            result_id: Analysis result ID.
+
+        Returns:
+            Analysis result data.
+        """
+        return self.base_api.analysis_result(result_id).get()
+
+    def device_components(self, backend_name: Optional[str]) -> List[Dict]:
```

Comment:
Seems like `backend_name` should not be optional.

Reply:
You don't want people to be able to query for all device components?

Reply:
I don't know what I device component means in the absence of a backend, but maybe someone will come up with a use case for wanting the set of all device components across all backends.